cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceTaskContext.java

```diff
@@ -85,6 +86,13 @@
   WorkflowToken getWorkflowToken();
 
   /**
+   * @return information about the enclosing {@link Workflow} run, if this {@link MapReduce} program is executed
+   * as a part of the Workflow; returns {@code null} otherwise. .
+   */
+  @Nullable
+  WorkflowInfo getWorkflowInfo();
```

Comment:
Now that we have `WorkflowInfo` interface, do you think its better to deprecate `MapReduceContext.getWorkflowToken` and have `getToken` method in the `WorkflowInfo` interface?